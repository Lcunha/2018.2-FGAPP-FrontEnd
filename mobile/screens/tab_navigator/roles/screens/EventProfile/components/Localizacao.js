import React, { Component } from "react"
import { Card, CardItem, Icon, Button } from "native-base"
import { View, ScrollView, Text, StyleSheet } from "react-native"
import { MapView, Location, Permissions } from "expo"
import { Marker, Circle } from "react-native-maps"

import CardHeader from "./CardHeader"

const latLngDelta = 0.005

class Localizacao extends Component {
	static navigationOptions = {
		title: "<MapView />"
	}

	state = {
		enderecoRole: "",
		region: {
			latitude: 0,
			longitude: 0,
			latitudeDelta: 0,
			longitudeDelta: 0
		},
		coordenadaRole: {
			//Valores Padrões para Latitude e Longitude. Servem para caso a informação dada de localização não seja válida
			latitude: this.props.latitude,
			longitude: this.props.longitude,
			latitudeDelta: this.props.latitudeDelta,
			longitudeDelta: this.props.longitudeDelta
		},
		alignLocalBtn: false
	}

	componentDidMount() {
    Permissions.askAsync(Permissions.LOCATION)
    this._getAddressFromCoordinates()
    this.setState({ region: this.state.coordenadaRole })
	}

	_onRegionChange(region) {
		this.setState({ region })
	}

	_getAddressFromCoordinates = async () => {
		try {
			let result = await Location.reverseGeocodeAsync({
				latitude: this.state.coordenadaRole.latitude,
				longitude: this.state.coordenadaRole.longitude
			})
			this.setState({ enderecoRole: result[0] })
		} catch (e) {
			this.setState({ error: e })
		}
  }

	render() {
		const { enderecoRole } = this.state

		return (
			<View>
				<Card>
					<CardHeader text="Localização" />

					<CardItem>
						<Icon name="pin" style={styles.pinIcon} />

						<View style={{ width: "100%", marginLeft: 20 }}>
							<Text style={{ fontWeight: "bold" }}>
								{this.props.placeName}
							</Text>
							<Text style={{ color: "grey" }}>
								{enderecoRole.city != null &&
									enderecoRole.city + ", "}
								{enderecoRole.street != null &&
									enderecoRole.street + ", "}
								{enderecoRole.name != null &&
									enderecoRole.name + "\n"}
								{enderecoRole.region != null &&
									enderecoRole.region + " - "}
								{enderecoRole.country != null &&
									enderecoRole.country + "\n"}
								{enderecoRole.postalCode != null &&
									enderecoRole.postalCode + "\n"}
							</Text>
						</View>
					</CardItem>

					{this.state.region.latitude != 0 && (
						<ScrollView style={styles.mapView}>
							<MapView
								ref={ref => {
									this._mapView = ref
								}}
								style={{
									width: "100%",
									height: 250
								}}
								region={this.state.region}
								onRegionChangeComplete={region =>
									this._onRegionChange(region)
								}
								loadingEnabled={true}
							>
								<Marker
									coordinate={{
										latitude: this.state.coordenadaRole
											.latitude,
										longitude: this.state.coordenadaRole
											.longitude
									}}
									title={this.props.placeName}
									description={"Local do Rolê"}
								/>
								<Circle
									center={{
										latitude: this.state.coordenadaRole
											.latitude,
										longitude: this.state.coordenadaRole
											.longitude
									}}
									radius={20}
									strokeWidth={2}
									strokeColor="green"
								/>
							</MapView>
						</ScrollView>
					)}
					<Text style={styles.coordinates}>
						Latitude: {this.state.region.latitude} / Longitude:{" "}
						{this.state.region.longitude}
					</Text>

					<Button
						onPress={() => {
							this._mapView.animateToCoordinate({
								latitude: this.state.coordenadaRole.latitude,
								longitude: this.state.coordenadaRole.longitude
							})
						}}
						block
						style={styles.centralizeBtn}
					>
						<Icon name="locate" style={{ color: "black" }} />
						<Text> Realinhar Localização</Text>
					</Button>
				</Card>
			</View>
		)
	}
}

export default Localizacao

const styles = StyleSheet.create({
	mapView: {
		borderWidth: 1,
		borderColor: "grey",
		width: "95%",
		alignSelf: "center"
	},
	coordinates: {
		color: "grey",
		textAlign: "center",
		alignSelf: "center",
		fontSize: 10,
		marginBottom: 10
	},
	pinIcon: {
		fontSize: 50,
		color: "black"
	},
	centralizeBtn: {
		marginBottom: 10,
		alignSelf: "center",
		width: "95%",
		backgroundColor: "white",
		borderWidth: 1,
		borderColor: "black",
		borderRadius: 20
	}
})
