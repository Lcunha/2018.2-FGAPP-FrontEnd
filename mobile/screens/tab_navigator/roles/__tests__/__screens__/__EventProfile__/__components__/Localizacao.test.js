import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Localizacao from '../../../../screens/EventProfile/components/Localizacao';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<Localizacao />).toJSON();
  expect(tree).toMatchSnapshot();
});
