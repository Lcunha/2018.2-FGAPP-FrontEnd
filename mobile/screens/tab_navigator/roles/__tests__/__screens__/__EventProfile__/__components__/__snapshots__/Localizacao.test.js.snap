// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "marginBottom": 10,
        }
      }
    >
      Localização
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <View
        style={
          Object {
            "marginLeft": 20,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "grey",
            }
          }
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "grey",
          "fontSize": 10,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Latitude: 
      0
       / Longitude:
       
      0
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderBottomWidth": null,
          "borderColor": "black",
          "borderLeftWidth": null,
          "borderRadius": 20,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": 1,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "width": "95%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         Realinhar Localização
      </Text>
    </View>
  </View>
</View>
`;
